### PR TITLE
Load Balancer migrate to ALB (ELBv2) + CodeDeploy-managed instance registration/de-registration during deployment

### PR DESCRIPTION
**Pull request by piotr-czerwik**

### CODE REVIEW

**Comment by spudtheimpaler** [comment]
lgtm

File: src/main/bundle/cloudformation-templates/web-cluster-template.json [hunk 1]
```diff
@@ -49,48 +50,52 @@
...
+      "Default": "1.18.210"
     },
-
     "ManagedPolicyArns": {
       "Description": "List of IAM policy names that will be granted to the instance role",
       "Type": "CommaDelimitedList",
       "Default": ""
     },
-
     "CreateELB": {
```
[the remark this answers is not in the document]
Comment (piotr-czerwik) [reply src/main/bundle/cloudformation-templates/web-cluster-template.json:76]: That was mine goal - reduce the number of required param changes.

File: src/main/bundle/cloudformation-templates/web-cluster-template.json [hunk 2]
```diff
@@ -463,100 +434,85 @@
...
-        "ConnectionDrainingPolicy": {
-          "Enabled": "true",
-          "Timeout": "300"
-        },
-        "ConnectionSettings": {
-          "IdleTimeout": "60"
-        },
-        "CrossZone": "true",
+      "Properties" : {
+        "Name": { "Fn::Join": ["", [{ "Ref": "AWS::StackName" }, "-TG"] ] },
```
[the remark this answers is not in the document]
Comment (piotr-czerwik) [reply src/main/bundle/cloudformation-templates/web-cluster-template.json:441]: The names have length limitations. I've found it more meaningful to display the fill stack name than waste additional 12 characters here.

File: src/main/bundle/cloudformation-templates/web-cluster-template.json [hunk 3]
```diff
@@ -463,100 +434,85 @@
...
-              }
-            ]
-          }
-        ]
+        "LoadBalancerArn": { "Ref": "MyALB" },
+        "Port": "443",
+        "Protocol": "HTTPS",
+        "Certificates": [
+          { "CertificateArn": { "Ref": "SSLCertificateId" } }
+        ],
```
[the remark this answers is not in the document]
Comment (piotr-czerwik) [reply src/main/bundle/cloudformation-templates/web-cluster-template.json:511]: Yes, that was my intention to do the migration first, as it fixes the ASG suspended processes issues we have.
How to support additional listeners requires a separate design discussion.

**Pull request merged**